### DISCUSSION TITLE
Quarkus 3 Beta1 Hibernate Reactive queries use class properties instead of column names?

### DISCUSSION THREAD

**Discussion by hakuseki** [quarkusio/quarkus#32160] [Q&A] [answered]
Hi

I'm trying out the Quarkus Beta1 release on my application.

In Quarkus 2 my class (as a PanacheEntity) has properties such as ```userName```, ```personNumber```, in camelCase but the annotation towards the column in PG is using lowercase, ```@Column(name = "username", updatable = false)``` …

When querying I use the following query:

```
return find(
                "((personnumber = :pnr) AND personnumber is not null) OR ((username = :userId) AND username is not " +
                "null and password = crypt(:pwd, password))",
                Parameters.with("pnr", pnr)
                          .and("userId", userId)
                          .and("pwd", pwd)).singleResult();
```

In Quarkus 3 Beta1 this is not working anymore.

I get the following error:

```
org.hibernate.query.SemanticException: Could not interpret path expression 'username'
```
or
```
org.hibernate.query.SemanticException: Could not interpret path expression 'personnumber'
```

But if I change the query to use the correct property name instead it's working fine.

```
return find(
                "((personNumber = :pnr) AND personNumber is not null) OR ((userName = :userId) AND userName is not " +
                "null and password = crypt(:pwd, password))",
                Parameters.with("pnr", pnr)
                          .and("userId", userId)
                          .and("pwd", pwd)).singleResult();
```

Is this an expected behaviour in the future or a result of the beta or might I have been doing the wrong way from the beginning but Hibernate was forgiving?

Disclaimer: my Hibernate knowledge is a bit of a novice…

[a comment by an automated account is not shown]

**Comment by DavideD** [comment, marked as answer]
When you write HQL, you are supposed to refer to the object property name (and not the name of the column on the database). 

With ORM 5, it worked anyway,  but the latest Hibernate Reactive is based on ORM 6.2 that's stricter.

**Reply by mklueh** [reply]
Nice to know. Not sure if I have misconfigured my IDE, but as I don't have autocomplete and type-safety within the find() method, I now use the Lombok `@FieldNameConstants` annotation to get typesafe fields and use them in the find method like

```
find(User.Fields.emailAddress + " :email", Parameters.with("email", email))
```

**Comment by hakuseki** [comment]
I understand, thanks!

I also had another query where I updated a timestamp which didn't work:
```
return update("migration_state = true, migrated_timestamp = now() where id = ?1 and migration_state = false",
                      id);
```

but that also failed now so with your explanation I had to do it like this:
```
return update("migrationState = true, migratedTimestamp = ?2 where id = ?1 and migrationState = false",
                      id, OffsetDateTime.now());
```

**Reply by DavideD** [reply]
I think it's for the same reason

`now()` is a function from Postgres, it's not HQL.

You can check what supported by HQL in the Hibernate ORM documentation: [16. Hibernate Query Language](https://docs.jboss.org/hibernate/orm/current/userguide/html_single/Hibernate_User_Guide.html#query-language)

I haven't tested it, but I think you could use the [sql function](https://docs.jboss.org/hibernate/orm/current/userguide/html_single/Hibernate_User_Guide.html#hql-function-sql):
```
return update("migration_state = true, migrated_timestamp = sql('now()') where id = ?1 and migration_state = false", id);
```
Keep in mind that the way you've rewritten it is db independent, while the original approach would only work for PG.
